src/main/java/gov/usgs/cida/pubs/transform/CrossrefTransformer.java

```diff
@@ -130,49 +124,72 @@ public void write(Object result) throws IOException {
 	/**
 	 * 
 	 * @param pub
-	 * @return true if result was written successfully, false otherwise.
 	 * @throws IOException 
 	 */
-	protected boolean writeResult(Publication<?> pub) throws IOException {
-		boolean success = false;
+	protected void writeResult(Publication<?> pub) throws IOException {
+		Map<String, Object> model = makeModel(pub);
+		boolean createMinimal = false;
+
+		// first try to write the full Crossref document
 		try {
 			LOG.trace("Writing crossref report entry for publication with indexId = '" + pub.getIndexId() + "'");
+			writeModelToTemplate(model, "crossref/body.ftlx");
+		} catch (TemplateException | IOException e) {
+			String logMessage = String.format("Error transforming publication (indexId: %s doi: %s) into Crossref XML, retrying with minimal crossref",
+								pub.getIndexId(), pub.getDoi());
+			LOG.error(logMessage, e);
+			createMinimal = true;
+		}
 
-			List<PublicationContributor<?>> contributors = this.getContributors(pub);
-			if (contributors.isEmpty()) {
-				String message = getExcludedErrorMessage(pub);
-				LOG.error(message + ". Publication had no contributors.");
-				writeComment(message);
-			} else {
-				Map<String, Object> model = new HashMap<>();
-				model.put("pub", pub);
-				boolean isNumberedSeries = PubsUtils.isUsgsNumberedSeries(pub.getPublicationSubtype());
-				model.put("isNumberedSeries", isNumberedSeries);
+		// if full document failed, try creating a minimal Crossref
+		if(createMinimal) {
+			String logMessage = "";
+			try {
+				model.put("reasonForMinRec", getMinimalRecordComment(pub));
```

Comment:
Yes, getMinimalRecordComment() duplicates checks the templating tool performs. However, this makes it way easier for a support person or user to see that the full record failed because onlineIssn was missing by simply looking at the document. Otherwise, someone with access to the logs has to search for the error, and parse out field name from the freemarker stack trace.

Reply:
Ideally we would use template exception getMessageWithoutStackTop() for the xml comment, that is is probably not quite user friendly. I am ok with taking it out, since ideally the, minimal record should just be a rare exception.